src/opendr/engine/target.py

```diff
@@ -68,6 +122,17 @@ def __init__(self, prediction: int, description=None, confidence=None):
         self.confidence = confidence
         self.description = description
 
+    @Target.data.setter
+    def data(self, data):
+        """
+        Setter for data. Category expects data of int type.
+        :param: data to be used for creating a Category object
+        """
+        if isinstance(data, int) or not data:
```

Comment:
This will actually allow other types than just integers. Did you mean to use `isinstance(data, int) or data is None`?

Reply:
Actually, this shouldn't be there in the first place (assignments with None do not make sense), so I completely removed it.